src/emulator/main.cpp

```diff
@@ -137,99 +97,31 @@ int main(int argc, char *argv[]) {
     }
 
     Ptr<const void> entry_point;
-    if (!load_app(entry_point, host, path, is_vpk)) {
-        std::string message = "Failed to load \"";
-        message += wide_to_utf(path);
-        message += "\"";
-        message += "\nSee console output for details.";
-        error(message.c_str(), host.window.get());
-        return ModuleLoadFailed;
-    }
+    if (auto err = load_app(entry_point, host, path, is_vpk) != Success)
+        return err;
 
-    const CallImport call_import = [&host](uint32_t nid, SceUID main_thread_id) {
-        ::call_import(host, nid, main_thread_id);
-    };
+    if (auto err = run_app(host, entry_point) != Success)
+        return err;
 
-    const SceUID main_thread_id = create_thread(entry_point, host.kernel, host.mem, host.io.title_id.c_str(), SCE_KERNEL_DEFAULT_PRIORITY_USER, SCE_KERNEL_STACK_SIZE_USER_MAIN, call_import, false);
-    if (main_thread_id < 0) {
-        error("Failed to init main thread.", host.window.get());
-        return InitThreadFailed;
-    }
-
-    const ThreadStatePtr main_thread = find(main_thread_id, host.kernel.threads);
-    Ptr<void> argp = Ptr<void>();
-    if (!strncmp(host.kernel.loaded_modules.begin()->second->module_name, "SceLibc", 7)) {
-        const SceUID libc_thread_id = create_thread(host.kernel.loaded_modules.begin()->second->module_start, host.kernel, host.mem, "libc", SCE_KERNEL_DEFAULT_PRIORITY_USER, SCE_KERNEL_STACK_SIZE_USER_DEFAULT, call_import, false);
-        const ThreadStatePtr libc_thread = find(libc_thread_id, host.kernel.threads);
-        run_on_current(*libc_thread, host.kernel.loaded_modules.begin()->second->module_start, 0, argp);
-        libc_thread->to_do = ThreadToDo::exit;
-        libc_thread->something_to_do.notify_all(); // TODO Should this be notify_one()?
-        host.kernel.running_threads.erase(libc_thread_id);
-        host.kernel.threads.erase(libc_thread_id);
-    }
-
-    if (start_thread(host.kernel, main_thread_id, 0, Ptr<void>()) < 0) {
-        error("Failed to run main thread.", host.window.get());
-        return RunThreadFailed;
-    }
-
-    GLuint TextureID = 0;
-    host.t1 = SDL_GetTicks();
+    GLuint fb_texture_id = 0;
+    host.sdl_ticks = SDL_GetTicks();
     while (handle_events(host)) {
-        if (!TextureID) {
-            glGenTextures(1, &TextureID);
```

Comment:
This looks like initialisation code which could just be moved up. I don't think it'll be possible to not have a texture.

Reply:
This was meant to be a purely-refactoring PR and I'm not sure if that's the case. Surely it's like this for a reason?

Reply:
No problem.